examples/growing_bacteria.jl

```diff
@@ -0,0 +1,185 @@
+using Agents
+using LinearAlgebra
+
+"""
+    SimpleCell <: AbstractAgent
+A simple bacterial cell, modelled by two soft disks linked with a spring.
+Based on an (as of 04/2020) unpublished model by Yoav G. Pollack and Philip Bittihn
+"""
+mutable struct SimpleCell <: AbstractAgent
+    id::Int
+
+    pos::NTuple{2, Float64}
+    length::Float64
+    orientation::Float64
+    growthprog::Float64
+
+    growthrate::Float64
+
+    # node positions/forces
+    p1::NTuple{2, Float64}
+    p2::NTuple{2, Float64}
+    
+    f1::NTuple{2, Float64}
+    f2::NTuple{2, Float64}
+
+    function SimpleCell(id, pos, l, φ, g, γ)
+        a = new(id, pos, l, φ, g, γ, (0.0, 0.0), (0.0, 0.0),  (0.0, 0.0), (0.0, 0.0))
+        update_nodes!(a)
+        return a
+    end
+end
+
+
+"""
+    update_nodes!(agent::SimpleCell)
+Updates the node positions `agent.p1` and `agent.p2` from the cell coordinates
+""" 
+function update_nodes!(a::SimpleCell)
+    offset = 0.5 * a.length .* unitvector(a.orientation)
+    a.p1 = a.pos .+ offset
+    a.p2 = a.pos .- offset
+end
+
+
+unitvector(φ) = reverse(sincos(φ))
+cross2D(a, b) = a[1]*b[2] - a[2]*b[1]    
+
+
+## API functions ###########################################################################
+
+function model_step!(model)
+    for a in allagents(model)
+        if a.growthprog ≥ 1
+            # split old cells into daughters
+            add_agent!(a.p1, model, 0.0, a.orientation, 0.0, 0.1*rand() + 0.05)
+            add_agent!(a.p2, model, 0.0, a.orientation, 0.0, 0.1*rand() + 0.05)
+            kill_agent!(a, model)
+        else
+            # compute internal spring compression
+            uv = unitvector(a.orientation)
+            internalforce = model.properties[:hardness]*(a.length - a.growthprog) .* uv
+            a.f1 = -1 .* internalforce
+            a.f2 = internalforce
+        end
+    end
+
+    for (a1, a2) in interacting_pairs(model, 2, :all)
+        interact!(a1, a2, model)
+    end
+end
+
+
+# use custom move_agent! because this model doesn't work well with defaults
+function agent_step!(agent::SimpleCell, model::ABM)
+    fsym, compression, torque = transform_forces(agent)
+    # overdamped dynamics, v ∝ F
+    agent.pos = agent.pos .+ model.dt*model.properties[:mobility].*fsym
```

Comment:
I haven't fully understood why `move_agent!` can't be used here. it seems to me that the agent's velocity simply becomes `model.properties[:mobility].*fsym`. Is this incorrect? Furthermore, the function `transform_forces` seems to only use the self-agent, which makes me "believe" that the `update_vel!` functionality is in fact possible. What do you think? If you don't want to try it, I'll try it when I find some extra time.

I don't think we can show something like `Agents.DBInterface.execute(model.space.updateq, (agent.pos..., agent.id))` in an example that is supposed to be exposed to the docs.

Reply:
Yeah, but there are three more "position-like" variables to be updated and I didn't like splitting what is essentially the same task (ie. `a = a + dt*b`) into two functions. 

I agree that the DBInterface should not be this visible in an example, but as there will probably be models that cannot use `move_agent!` for some reason but still need to update the DB, it might be useful to wrap this statement into a one-line function.

Reply:
Okay, good point, I'll get around to that.